lib/user.rb

```diff
@@ -0,0 +1,31 @@
+class User
+  attr_reader :name, :jokes
+
+  def initialize(name, jokes = [])
+    @name = name
+    @jokes = jokes
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(user, joke)
+    user.learn(joke)
+  end
+
+  def joke_by_id(id)
+    correct_joke = []
+
+    @jokes.each do |joke|
+      if joke.id == id
+      correct_joke << joke
+      end
+      correct_joke
+    end
+
+    correct_joke.first
+
+  end
```

Comment:
This works, but there's a good enumerable to search for just one of something that uses the same boolean expression that you have on line 21, which would reduce this whole method to one line if you use `{}` instead of do/end.